test/file.ts

```diff
@@ -3853,6 +3865,129 @@ describe('File', () => {
+      retryCount: number;
+      constructor(retryCount: number) {
+        super();
+        this.retryCount = retryCount;
+      }
+      _transform(chunk: any, _encoding: string, done: Function) {
+        this.push(chunk);
+        setTimeout(() => {
+          if (this.retryCount === 1) {
+            done(new HTTPError('first error', 500));
+          } else {
+            done();
+          }
+        }, 5);
+      }
+    }
+
+    describe('retry mulipart upload', () => {
+      it('should save a string with no errors', async () => {
+        const options = {resumable: false};
+        file.createWriteStream = (options_: {}) => {
+          return new DelayedStreamNoError();
+        };
+        await file.save(DATA, options, assert.ifError);
+      });
+
+      it('string upload should retry on first failure', async () => {
+        const options = {resumable: false};
+        let retryCount = 0;
+        file.createWriteStream = (options_: {}) => {
+          retryCount++;
+          return new DelayedStream500Error(retryCount);
+        };
+        await file.save(DATA, options);
+        assert.ok(retryCount === 2);
+      });
+
+      it('string upload should not retry if nonretryable error code', async () => {
+        const options = {resumable: false};
+        let retryCount = 0;
+        file.createWriteStream = (options_: {}) => {
+          class DelayedStream403Error extends Transform {
+            _transform(chunk: any, _encoding: string, done: Function) {
+              this.push(chunk);
+              setTimeout(() => {
+                retryCount++;
+                if (retryCount === 1) {
+                  done(new HTTPError('first error', 403));
+                } else {
+                  done();
+                }
+              }, 5);
+            }
+          }
+          return new DelayedStream403Error();
+        };
+        try {
+          await file.save(DATA, options);
+          throw Error('unreachable');
+        } catch (e) {
+          assert.strictEqual(e.message, 'first error');
+        }
+      });
+
+      it('should save a buffer with no errors', async () => {
+        const options = {resumable: false};
+        file.createWriteStream = (options_: {}) => {
+          return new DelayedStreamNoError();
+        };
+        await file.save(DATA, options, assert.ifError);
+      });
+
+      it('buffer upload should retry on first failure', async () => {
+        const options = {resumable: false};
+        let retryCount = 0;
+        file.createWriteStream = (options_: {}) => {
+          retryCount++;
+          return new DelayedStream500Error(retryCount);
+        };
+        await file.save(BUFFER_DATA, options);
+        assert.ok(retryCount === 2);
+      });
+
+      it('non-multipart upload should not retry', async () => {
+        const options = {resumable: true};
+        let retryCount = 0;
+        file.createWriteStream = (options_: {}) => {
```

Comment:
Can you just get rid of the `option_` if you're not using it? alternatively putting a leading `_` rather than trailing `_` might satisfy the linter.